modules/renter/hostdb/hostdb.go

```diff
@@ -119,6 +123,17 @@ func newHostDB(g modules.Gateway, cs modules.ConsensusSet, persistDir string, de
 	if err != nil && !os.IsNotExist(err) {
 		return nil, err
 	}
+	hdb.tg.AfterStop(func() {
+		hdb.mu.Lock()
+		err := hdb.saveSync()
+		hdb.mu.Unlock()
```

Comment:
just wondering, is it necessary to lock in `AfterStop` functions? In theory this should be the only goroutine running, right?